src/perception/cost_map/cost_map.processing.cpp

```diff
@@ -0,0 +1,62 @@
+#include "cost_map.hpp"
+#include "../point.hpp"
+
+#include "loop_profiler.hpp"
+#include <Eigen/src/Core/Matrix.h>
+#include <Eigen/src/Core/util/ForwardDeclarations.h>
+#include <cstdint>
+#include <pstl/glue_execution_defs.h>
+#include <regex>
+
+namespace mrover {
+
+    void CostMapNodelet::pointCloudCallback(sensor_msgs::PointCloud2ConstPtr const& msg) {
+        assert(msg);
+        assert(msg->height > 0);
+        assert(msg->width > 0);
+
+        if (!mPublishCostMap) return;
+        try {
+            SE3 zed_to_map = SE3::fromTfTree(tf_buffer, "map", "zed2i_left_camera_frame");
+            auto* points = reinterpret_cast<Point const*>(msg->data.data());
+            int step = 4;
+            uint32_t num_points = msg->width * msg->height / step;
+            Eigen::MatrixXd point_matrix(4, num_points);
+            Eigen::MatrixXd normal_matrix(4, num_points);
+            // std::for_each(points, points + msg->width * msg->height, [&](auto* point) {
+            for (auto point = points; point - points < msg->width * msg->height; point += step) {
+                Eigen::Vector4d p{point->x, point->y, 0, 1};
```

Comment:
Im pretty sure you should leave the z component in the point here and only set it to zero once its in the map frame, since if the zed is rotated you end up removing parts of the true x and y components

Reply:
This makes sense. Should be fixed now.